athena-redis/src/main/java/com/amazonaws/athena/connectors/redis/lettuce/RedisConnectionFactory.java

```diff
@@ -38,19 +38,33 @@
   private static final int CONNECTION_TIMEOUT_MS = 2_000;
   private final Map<String, RedisConnectionWrapper<String, String>> clientCache = new HashMap<>();
 
-  public synchronized RedisConnectionWrapper<String, String> getOrCreateConn(String conStr, boolean sslEnabled,
+  public synchronized RedisConnectionWrapper<String, String> getOrCreateConn(String conStr,
+                                                                             boolean sslEnabled,
                                                                              boolean isCluster)
+  {
+    return getOrCreateConn(conStr, sslEnabled, isCluster, "0");
+  }
+
+  public synchronized RedisConnectionWrapper<String, String> getOrCreateConn(String conStr,
+                                                                             boolean sslEnabled,
+                                                                             boolean isCluster,
+                                                                             String dbNumber)
   {
     String conKey = conStr + sslEnabled + isCluster;
+    if (!isCluster && dbNumber != null && !"null".equals(dbNumber)) {
```

Comment:
I think if we get rid of the `.valueOf` we can get rid of this string equality check for "null"

Reply:
removed .valueOf and "null" eq check